src/controllers/controller.bar.js

```diff
@@ -355,6 +359,10 @@ module.exports = DatasetController.extend({
 			}
 		}
 
+		if (!stacks.length) {
+			stacks.push(undefined);
```

Comment:
Why is this required?

Reply:
So empty dataset will end up with one stack and avoid inf due to div0

Reply:
It seems potentially confusing to me that `stackCount` would sometimes be altered. Can we instead fix it where we do the division? E.g. change `size / ruler.stackCount` to `size / (ruler.stackCount || 1)`

Reply:
That was my first try. It did not solve the problem alone through and to me it seems logical that there is at least one stack/bar position per tick.

Reply:
I don't love it, but I suppose it's okay. Can you add a code comment explaining it?

Another option, could you change `getStackCount` to return 1 if there are no stacks? That maybe would be cleaner than adding an `undefined` stack

Reply:
Would need to make `getStackIndex` return `0` instead of `-1`, and that might just be wrong in some situations.

Reply:
Added a comment, and a test to catch regression.